### PR TITLE
Expand materials on functions

### PR DESCRIPTION
**Pull request by encukou**
The students often have issues with using functions. I've been trying out some changes:
- Use `len` rather than `sin` and get rid of `π`
- Explain expression evaluation (I probably should moved this before the functions lesson, though) 
- Spend some time “procedures” – functions that don't return an useful value. The distinction will hopefully help them with choosing `print(x)`/`return x` in the future.

If some of this is useful, incorporate it into your course and let me know how it works.

Fixes: https://github.com/pyvec/naucse.python.cz/pull/549

### CODE REVIEW

**Comment by encukou** [comment]
Split into 3 lessons after feedback on the sprint.

File: lessons/beginners/basic-functions/index.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
+
+# Užitečné funkce
+
+Ukažme si pár základních funkcí, které Python nabízí.
+
+Tato kapitola ukazuje výběr z nástrojů, které ti Python nabízí.
+Tvůj cíl není naučit se vše nazpaměť, ale mít přehled o tom, co je zhruba
+možné.
+Detaily můžeš vždycky dohledat – ať už na taháku (které jsou, na rozdíl od
+školy, vždycky povoleny!), v těchto matreriálech, nebo v oficiální
```
Comment (janpipek) [inline lessons/beginners/basic-functions/index.md]:
```suggestion
školy, vždycky povoleny!), v těchto materiálech, nebo v oficiální
```

File: lessons/beginners/basic-functions/index.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
+
+# Užitečné funkce
+
+Ukažme si pár základních funkcí, které Python nabízí.
+
+Tato kapitola ukazuje výběr z nástrojů, které ti Python nabízí.
```
Comment (janpipek) [inline lessons/beginners/basic-functions/index.md]:
```suggestion
Tato kapitola ukazuje výběr z nástrojů, které jsou ti v Pythonu k dispozici.
```
Výraz se opakuje

File: lessons/beginners/basic-functions/index.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+možné.
+Detaily můžeš vždycky dohledat – ať už na taháku (které jsou, na rozdíl od
+školy, vždycky povoleny!), v těchto matreriálech, nebo v oficiální
+dokumentaci či jinde na Internetu.
+
+Můžeš si stáhnout i
+<a href="https://github.com/encukou/cheatsheets/raw/master/basic-functions/basic-functions-cs.pdf">tahák</a>,
+který se rozdává na srazech.
+Doporučuji ho mít ze začátku při ruce.
+Když narazíš na úkol, který nevíš jak vyřešit, projdi si ho a zamysli se,
```
Comment (janpipek) [inline lessons/beginners/basic-functions/index.md]:
```suggestion
Když narazíš na úkol, se kterým si nevíš rady, projdi si tahák a zamysli se,
```

File: lessons/beginners/basic-functions/index.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+která z funkcí by se dala použít.
+
+
+## Vstup a výstup
+
+Tyhle funkce už známe.
+`print` vypíše nepojmenované argumenty, oddělené mezerou.
+Pojmenovaný argument `end` určuje co se vypíše na konci (místo přechodu
```
Comment (janpipek) [inline lessons/beginners/basic-functions/index.md]:
```suggestion
Pojmenovaný argument `end` určuje, co se vypíše na konci (místo přechodu
```

File: lessons/beginners/basic-functions/index.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+na nový řádek);
+`sep` zase co se vypíše mezi jednotlivými argumenty (místo mezery).
```
Comment (janpipek) [inline lessons/beginners/basic-functions/index.md]:
```suggestion
`sep` udává, co se vypíše mezi jednotlivými argumenty (místo mezery).
```

File: lessons/beginners/basic-functions/index.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+> [note]
+> Příklad opět spusť ze souboru, ne interaktivně:
+
+```python
+print(1, 'dvě', False)
+print(1, end=' ')
+print(2, 3, 4, sep=', ')
+```
+
+Základní funkce na načtení vstupu, `input`,
```
Comment (janpipek) [inline lessons/beginners/basic-functions/index.md]:
```suggestion
Základní funkce pro načtení vstupu, `input`,
```

File: lessons/beginners/basic-functions/index.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+float('blablabla')  # chyba!
+int('8.9')          # chyba!
+```
+
+…a jak si poradit s chybou, která nastane,
+když použiješ špatnou hodnotu, si řekneme později.
+
+### Převádění a `input`
+
+Převádění typů se často používá při načítání vstupu, třeba jako:
```
Comment (janpipek) [inline lessons/beginners/basic-functions/index.md]:
```suggestion
Převádění typů se často používá při načítání vstupu, třeba takto:
```

File: lessons/beginners/expressions/index.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
+# Vyhodnocování výrazů
+
+Už víš, že Python se dá použít jako kalkulačka: dokáže spočítat
+hodnotu *výrazu* (angl. *expression*) jako `3 * (5 + 2)`.
```
Comment (janpipek) [inline lessons/beginners/expressions/index.md]:
```suggestion
hodnotu *výrazu* (angl. *expression*) typu `3 * (5 + 2)`.
```

File: lessons/beginners/expressions/index.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+Jak to ale vlastně dělá?
+Jak se vyhodnocují výrazy?
+
+Pro základní výrazy je to tak, jak to možná znáš ze školy.
+U `3 * (5 + 2)`, nejdřív spočítáš to, co je v závorkách: `(5 + 2)` je `7`.
```
Comment (janpipek) [inline lessons/beginners/expressions/index.md]:
```suggestion
U `3 * (5 + 2)` nejdřív spočítáš to, co je v závorkách: `(5 + 2)` je `7`.
```

File: lessons/beginners/expressions/index.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+#          ╰──┬──╯
+vysledek =    9    / 4
+#             ╰────┬─╯
+vysledek =        2.25
+```
+
+Funguje to i u složitých výrazů.
+Python se složitými výrazy nemá problém.
+Jen člověk, který program čte či píše, se v nich může lehce ztratit.
+Když opravdu potřebuješ napsat složitý výraz, je dobré jej rozdělit na několik
```
Comment (janpipek) [inline lessons/beginners/expressions/index.md]:
```suggestion
Když opravdu potřebuješ napsat složitý výraz, nebude na škodu jej rozdělit na několik
```

File: lessons/beginners/expressions/index.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+menších nebo vysvětlit pomocí komentáře.
+
+Je ale dobré mít povědomí o tom, jak složité výrazy „fungují“,
+aby ses jich nemusel{{a}} bát.
+Měl{{a}} bys být schopn{{gnd('ý', 'á')}} vysvětlit, co se stane,
+když se Pythonu zeptáš kolik je -<var>b</var> + (<var>b</var>² +
```
Comment (janpipek) [inline lessons/beginners/expressions/index.md]:
```suggestion
když se Pythonu zeptáš, kolik je -<var>b</var> + (<var>b</var>² +
```

File: lessons/beginners/expressions/index.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+Výrazy se používají na více místech Pythonu než jen v přiřazování
+do proměnných.
+Třeba podmínka u `if` je taky výraz a vyhodnocuje se stejně jako ostatní
+výrazy:
+
+```python
+strana = -5
+
+if strana <= 0:
+    print("Příliš")
```
Comment (janpipek) [inline lessons/beginners/expressions/index.md]: Příliš... něco specifičtějšího?

File: lessons/beginners/basic-functions/index.md [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+```python
+from math import sin, cos, tan, sqrt, floor, ceil
+```
+
+Naimportované funkce pak můžeš použít:
+
+```python
+sin(uhel)       # sinus
+cos(uhel)       # kosinus
+tan(uhel)       # tangens
```
Comment (lessay) [inline lessons/beginners/basic-functions/index.md]: Poznamenal bych tu, že jde o úhel v radianech, nikoliv ve stupních.
Případně i jak to převést.

File: lessons/beginners/functions/index.md [hunk 14]
```diff
@@ -1,426 +1,302 @@
...
 
-### Nápověda
-
-Další funkce pomáhá programátorům:
-Můžeš si přímo z programu (nebo z interaktivního
-režimu) vyvolat nápovědu k nějaké funkci.
-(Občas bývá srozumitelná i pro začátečníky,
-občas bohužel spíš ne – v takovém případě zkus
-Google).
+Dá změnit i to, co `print` udělá na konci výpisu.
```
Comment (lessay) [inline lessons/beginners/functions/index.md]:
```suggestion
Dá se změnit i to, co `print` udělá na konci výpisu.
```

File: lessons/beginners/basic-functions/index.md [hunk 15]
```diff
@@ -0,0 +1,213 @@
...
+
+Tato kapitola ukazuje výběr z nástrojů, které ti Python nabízí.
+Tvůj cíl není naučit se vše nazpaměť, ale mít přehled o tom, co je zhruba
+možné.
+Detaily můžeš vždycky dohledat – ať už na taháku (které jsou, na rozdíl od
+školy, vždycky povoleny!), v těchto matreriálech, nebo v oficiální
+dokumentaci či jinde na Internetu.
+
+Můžeš si stáhnout i
+<a href="https://github.com/encukou/cheatsheets/raw/master/basic-functions/basic-functions-cs.pdf">tahák</a>,
```
Comment (frenzymadness) [inline lessons/beginners/basic-functions/index.md]: Pyvec repository instead of yours?

File: lessons/beginners/functions/index.md [hunk 16]
```diff
@@ -1,426 +1,302 @@
...
+Normálně přejde na nový řádek, ale argumentem `sep` můžeš říct co se má vypsat
```
Comment (frenzymadness) [inline lessons/beginners/functions/index.md]:
```suggestion
Normálně přejde na nový řádek, ale argumentem `end` můžeš říct, co se má vypsat 
```

**Comment by frenzymadness** [comment]
Very nice. I am sure that it'll help a lot to a better understanding of how a function call works and how it can be combined with other parts of the language. Good job!

File: lessons/beginners/basic-functions/index.md (hunk 15 above)
Comment (encukou) [reply lessons/beginners/basic-functions/index.md]:
```suggestion
<a href="https://github.com/pyvec/cheatsheets/raw/master/basic-functions/basic-functions-cs.pdf">tahák</a>,
```

File: lessons/beginners/expressions/index.md (hunk 8 above)
Comment (encukou) [reply lessons/beginners/expressions/index.md]: Tohle bych neměnil.

File: lessons/beginners/expressions/index.md (hunk 10 above)
Comment (encukou) [reply lessons/beginners/expressions/index.md]: Proč?

File: lessons/beginners/expressions/index.md (hunk 12 above)
Comment (encukou) [reply lessons/beginners/expressions/index.md]:
```suggestion
    print("Strana musí být kladná!")
```

File: lessons/beginners/basic-functions/index.md (hunk 13 above)
Comment (encukou) [reply lessons/beginners/basic-functions/index.md]: Poznamenáno. Přijde mi toho tam příliš, takže časem bych to radši odstranil a nechal jen na taháku, ale předtím budu muset kouknout co na to navazuje.

**Comment by encukou** [comment]
Thank you for the reviews!

**Comment by lessay** [comment]
@encukou Thank you for your work on this. Functions are one of the most important things in programming and there were always problems in understanding them.

**Comment by encukou** [comment]
You're welcome! I look forward to people testing the new explanations on people :)

**Pull request merged**